tockloader-lib/src/tabs/tab.rs

```diff
@@ -75,6 +76,52 @@ impl Tab {
         }
     }
 
+    pub fn filter_tbfs(
+        &self,
+        settings: &BoardSettings,
+    ) -> Result<Vec<(&TbfFile, u64, u64)>, TockloaderError> {
```

Comment:
```suggestion
    /// ...
    pub fn compatible_tbfs(
        &self,
        settings: &BoardSettings,
    ) -> impl Iterator<item = TbfFile> { // Maybe Item = &TbfFile, unsure, play around
```

You can then just make this whole function a big old `map`

```rust
self.tbf_files.iter().map(|tbf|
    if !tbf.filename.arch == settings.arch.as_ref().unwrap() {
        return None;
    }
 
    match (tbf.filename.flash_address, tbf.filename.ram_address) {
        (None, None) => Some(tbf),
        (Some(flash), Some(ram)) => { ... some if flash >= and ram >=, otherwise None }
        _ => unreachable!("TbfFilename should not have only ram or only flash address speecified")
    }
```